packages/core/src/storage/Repository.ts

```diff
@@ -66,6 +66,21 @@ export class Repository<T extends BaseRecord<any, any, any>> {
     })
   }
 
+  /**
+   * Delete record by id. Returns null if no record is found
+   * @param id the id of the record to delete
+   * @returns
+   */
+  public async deleteById(id: string): Promise<void | null> {
```

Comment:
I guess this `<void | null>` comes from the pattern followed in `findById`, where we wanted to accept returning record as null to not throw an exception. This is the case as well?